bifrost/app/blog/blogs/test-your-llm-prompts/src.mdx

```diff
@@ -0,0 +1,286 @@
+
+---
+
+# How to properly test your prompts
+
+Properly testing your prompts involves setting up a systematic workflow that iteratively improves performance. This process ensures you’re equipped to handle dynamic scenarios, minimize errors, and optimize user experience.
+
+![Prompt Evaluation Lifecycle in Helicone](/static/blog/test-your-llm-prompts/prompt-evaluation-lifecycle.webp)
+_Steps to test your prompt: log > evaluate > experiment > deploy_
+
+## Preparation
+
+### **Step 1: Log your LLM requests**
+
+Use an observability tool to log your LLM requests and track key metrics like usage, latency, cost, time-to-first-token (TTFT). These tools provide dashboards to help you monitor irregularity, such as:
+
+- Rising error rates
+- Sudden spikes in API costs.
+- Declining user satisfaction scores.
+
+This data helps you identify when it’s time to improve your prompt.
+
+## Testing Process
+
+### Step 2: Create prompt variations
+
+Experiment with prompt versions using <a href="https://www.helicone.ai/blog/prompt-engineering-tools" target="_blank" rel="noopener noreferrer">techniques like chain-of-thought reasoning and multi-shot prompting</a>. Testing environments like _Helicone_ makes it easier to track prompt versions, inputs, and outputs, while also providing rollback capabilities if changes lead to underperformance/regression.
+
+<BottomLine
+  title="💡 Pro Tip"
+  description="You can create as many prompt variations as you need, where each tests one variable at a time (i.e. phrasing, model parameters) to isolate its impact."
+/>
+
+### Step 3: Use real data to generate outputs
+
+Run your prompts on real-world datasets to ensure they can handle variability and edge cases. There are 2 options:
+
+- **Golden datasets**: Curated inputs with known expected outputs.
+- **Randomly sampled production data**: More representative of real-world scenarios. Here's <a href="https://www.helicone.ai/blog/prompt-evaluation-for-llms" target="_blank" rel="noopener noreferrer">why this approach is better.</a>
+
+### Step 4: Compare outputs
+
+Evaluate the performance of your prompts using the best methods that suits your goals and capacity, such as:
+
+- **Real user feedback** for subjective insights.
+- **Human evaluators** for nuanced assessments.
+- **LLM-as-a-judge** for scalable and efficient comparisons.
+
+### Step 5: Push the best prompt to production
+
+Once you've identified the best-performing prompt, deploy it to production. Remember to continue to monitor your application using observability tools to track metrics and identify opportunities for further refinement.
+
+<BottomLine
```

Comment:
Not too sure what this means.

Reply:
Ok, segmenting "users" can make sense but this is a bit confusing. Honestly probably easiest to just remove.